gusto/linear_solvers.py

```diff
@@ -37,8 +41,9 @@ def __init__(self, state, solver_parameters=None,
                 solver_parameters = p
             self.solver_parameters = solver_parameters
 
-        # setup the solver
-        self._setup_solver()
+    @cached_property
+    def _solvers(self):
+        pass
```

Comment:
I think you can write this as an `@abstractproperty` to indicate the the subclass must provide it?

Reply:
instead of `@cached_property`?

Reply:
You can document this as `@abstractproperty`, and then all subclass implementations can mark the method as `@cached_property`. The only thing `@abstractproperty` will do is raise an exception if a subclass fails to implement a `_solvers` method.

Reply:
ah, I see - thanks @thomasgibson !